packages/core/src/browser/frontend-connection-status.ts

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, optional } from 'inversify';
+import { Endpoint } from './endpoint';
+import { ILogger } from '../common/logger';
+import { AbstractDialog } from './dialogs';
+import { Event, Emitter } from '../common/event';
+import { MessageService } from '../common/message-service';
+import { StatusBar, StatusBarAlignment } from './status-bar/status-bar';
+import { FrontendApplicationContribution, DefaultFrontendApplicationContribution } from './frontend-application';
+
+/**
+ * Service for listening on backend connection changes.
+ */
+export const ConnectionStatusService = Symbol('ConnectionStatusService');
+export interface ConnectionStatusService {
+    readonly onStatusChange: Event<ConnectionStatusChangeEvent>;
```

Comment:
I think getter is missing for a current state in the case if someone wants to subscribe after the first event was fired.